5.17.0-dev - linking error - undefined reference to `video::OGLBufferObject::~OGLBufferObject()'

### Luanti version

```true
Luanti 5.17.0-7701e71 (Linux)
Using LuaJIT 2.1
Built by LCC 1.29.16 (gcc version 11.3.0 compatible)
Running on Linux/6.1 e2k
BUILD_TYPE=Release
```

### Operating system and version

OSL-9.4.7

### CPU model

Elbrus-8C

### GPU model

AMD Radeon RX 6400

### Active renderer

_No response_

### Summary

Hello. I get an error when linking:
```
/usr/bin/ld: ../irr/src/libIrrlichtMt.a(Driver.cpp.o): в функции «video::COpenGL3DriverBase::COpenGL3DriverBase(SIrrlichtCreationParameters const&, io::IFileSystem*, video::IContextManager*)»:
Driver.cpp:(.text+0x138c): undefined reference to `video::OGLBufferObject::~OGLBufferObject()'
/usr/bin/ld: Driver.cpp:(.text+0x1408): undefined reference to `video::OGLBufferObject::~OGLBufferObject()'
```

### Steps to reproduce

```
mkdir build_e8c_test && cd build_e8c_test
cmake -DCMAKE_BUILD_TYPE=Release -DDEVELOPMENT_BUILD=FALSE -DBUILD_CLIENT=TRUE -DBUILD_SERVER=TRUE -DRUN_IN_PLACE=FALSE -DBUILD_UNITTESTS=FALSE -DENABLE_GLES2=TRUE ..
make -j8
```

--- a/irr/src/OpenGL/BufferObject.h
+++ b/irr/src/OpenGL/BufferObject.h
@@ -21,7 +21,10 @@ public:
 	/// @note does not create on GL side
 	OGLBufferObject(Target target) : m_target(target) {}
 	/// @note does not free on GL side
-	~OGLBufferObject() = default;
+	~OGLBufferObject() {}
+	// ^ Do not use `= default;`. These should be equivalent here, but `= default`
+	// results in a linking error for obscure configurations
+	// (Edison Design Group C++ frontend + MCST LCC 1.29.16 and similar), see #17232.
 
 	/// @return "name" (ID) of this buffer in GL
 	GLuint getName() const { return m_name; }

